recipes/fmt/all/conanfile.py

```diff
@@ -42,13 +41,19 @@ def _has_with_os_api_option(self):
     def generate(self):
         if not self.options.header_only:
             tc = CMakeToolchain(self)
-            tc.generate()  
+            tc.cache_variables["FMT_DOC"] = False
+            tc.cache_variables["FMT_TEST"] = False
+            tc.cache_variables["FMT_INSTALL"] = True
+            tc.cache_variables["FMT_LIB_DIR"] = "lib"
+            if self._has_with_os_api_option:
+                tc.cache_variables["FMT_OS"] = bool(self.options.with_os_api)
+            tc.generate()
 
     def layout(self):
-        cmake_layout(self)
+        cmake_layout(self, src_folder="src")
 
     def config_options(self):
-        if self.settings.os == "Windows":
+        if self.settings.os == "Windows" or self.options.shared:
```

Comment:
```suggestion
        if self.settings.os == "Windows":
```
Config options does not parse options.